README.md

```diff
@@ -11,44 +11,69 @@
 
-        [agent]
-        report = true
-        pluginsync = true
+  ```
+[agent]
+report = true
+pluginsync = true
+  ```
 
-2. On the Puppet master, make sure the [`reports`](https://docs.puppetlabs.com/references/4.2.latest/configuration.html#reports) setting in the master section includes tagmail:
+2. On the Puppet master, include tagmail in the  [`reports`](https://docs.puppetlabs.com/references/4.2.latest/configuration.html#reports) setting in the master section:
 
-        [master]
-        reports = tagmail
+  ```
+[master]
+tagmap = $confdir/tagmail.conf
+reports = puppetdb,console,tagmail
+  ```
 
-3. If you use anti-spam controls such as greylisting on your mail server, be sure to whitelist Puppet's sending email address to ensure your tagmail reports are not discarded as spam.
+3. If you use anti-spam controls such as grey-listing on your mail server, whitelist Puppet's sending email address to ensure your tagmail reports are not discarded as spam. This setting is controlled by the `reportfrom` setting in `puppet.conf`.
 
-4. In your Puppet confdir on your master, create a tagmail.conf file. This file will contain your email transport config options, as well as the tags themselves.
+4. In the Puppet confdir on your master, create a `tagmail.conf` file. This file will contain your email transport config options, as well as the tags themselves.
 
 ## Usage
 
-To configure the tagmail module, edit (or create and edit) the `tagmail.conf` in your Puppet confdir.
+### Tags
+
+Tags let you set context for resources, classes, and defined types. For example, you can assign a tag to all resources associated with a particular operating system, location, or other characteristic. The tag is then included in all log messages related to those resources.
+
+Puppet's [loglevels](https://docs.puppet.com/latest/metaparameter.html#loglevel) (`debug`, `info`, `notice`, `warning`, `err`, `alert`, `emerg`, `crit`, and `verbose`) can also be used as tags, and the `all` tag always matches every log message. To learn more about tags, see [tags](http://docs.puppetlabs.com/puppet/latest/reference/lang_tags.html) in the Puppet Language docs.
+
+### Configure `tagmail.conf`
+
+To configure the tagmail module, edit the `tagmail.conf` file you created in Step 4 above. This file is located in your Puppet confdir. The `tagmail.conf` should be formatted as an ini file.
+
+1. Open `tagmail.conf` in your text editor and add create `[transport]` and `[tagmap]` sections.
+
+1. In the `[transport]` section, specify _either_:
+
+  * `sendmail`, with a path to your sendmail binary (by default, `/usr/sbin/sendmail`).
+  * `smtpserver`, `smtpport`, and `smtphelo`. If you do not specify `smtpserver`, tagmail defaults to using `sendmail`.
+
+1. In the `[tagmap]` section , specify tags and email addresses. Each line should include both:
 
```

Comment:
numbering is weird in the above section. the raw md says 1. 1. 1. but when i look at the rich preview its 1. 2. 1.